src/Forms/GridField/GridField.php

```diff
@@ -444,6 +451,72 @@ private function initState(): void
         }
     }
 
+    /**
+     * Adds state for this gridfield from the request variables.
+     *
+     * If there is state already set on this GridField, that takes precedent
+     * over state from the request.
+     */
+    private function addStateFromRequest(): void
+    {
+        $request = $this->getRequest();
+        if (($request instanceof NullHTTPRequest) && Controller::has_curr()) {
+            $request = Controller::curr()->getRequest();
+        }
+        
+        $stateStr = $this->getStateManager()->getStateFromRequest($this, $request);
+
+        if ($stateStr) {
+            $oldState = $this->getState(false);
+            // Create a dummy state so that we can merge the current state with the request state.
+            $newState = new GridState($this, $stateStr);
+            // Put the current state on top of the request state.
+            $newState->setValue($oldState->Value());
+            $this->state = $newState;
+        }
+    }
+
+    /**
+     * Add get and post parameters if exist to the URL.
+     * URL keeps state for GridField, that exist in few different tabs.
+     */
+    public function addOtherGridfieldStateToURL(string $link, bool $addState = true): string
```

Comment:
I know I asked you to change it to this, but that was before we started adding the _current_ gridfield state as well... this name no longer makes sense because it's no longer _only_ other gridfield state. Sorry about that.
```suggestion
    public function addStateToURL(string $link, bool $addState = true): string
```

Reply:
Probably, we should use another name. Like addStateFromRequestToURL, because we have addStateToURL() in GridFIeldStateManager and also call this method here. It's a little bit confusing.

Reply:
It's not _only_ adding state from the request though, it's also adding state from the session (which, admittedly, _originates_ with the request...)

It's not uncommon for a method in classA to call a method in classB with the same name - in this case the state manager adds some state the to URL, and the gridfield adds some state to the URL - they're both just adding state to the URL. That said, I'm not against having some more specific name here if we can find one we both agree makes sense.
Maybe `addAllStateToUrl()`? Since it's adding state for all gridfields present in the request?

Reply:
DONE